src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/GroupChatResource.java

```diff
@@ -70,6 +72,9 @@ public GroupChatResource(UserRepository userRepository, CourseRepository courseR
     public ResponseEntity<GroupChatDTO> startGroupChat(@PathVariable Long courseId, @RequestBody List<String> otherChatParticipantsLogins) throws URISyntaxException {
         var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
         log.debug("REST request to create group chat in course {} between: {} and : {}", courseId, requestingUser.getLogin(), otherChatParticipantsLogins);
+        if (!(courseRepository.isMessagingEnabled(courseId))) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Messaging is not enabled for this course");
+        }
```

Comment:
```suggestion
        checkMessagingEnabledElseThrow(courseId);
```

Reply:
Refactored it to reduce code duplication